src/Tablebot/Plugins/Quote.hs

```diff
@@ -360,8 +379,52 @@ Calling without arguments returns a random quote
 quotePlugin :: Plugin
 quotePlugin =
   (plug "quote")
-    { commands = [quote],
+    { commands = [quote, commandAlias "q" quote],
       onReactionAdds = [quoteReactionAdd],
       migrations = [quoteMigration],
       helpPages = [quoteHelp]
     }
+
+deriving instance Generic Quote
+
+instance FromJSON Quote
+
+instance ToJSON Quote
+
+-- | Get all the quotes in the database.
+allQuotes :: DatabaseDiscord [Quote]
+allQuotes = fmap entityVal <$> selectList [] []
+
+-- | Export all the quotes in the database to either a default quotes file or to a given
+-- file name that is quoted in the command. Superuser only.
+exportQuotes :: Command
+exportQuotes = Command "export" (parseComm exportQ) []
+
+exportQ :: Maybe (Quoted FilePath) -> Message -> DatabaseDiscord ()
+exportQ qfp m = requirePermission Superuser m $ do
+  let defFileName = getSystemTime >>= \now -> return $ "quotes_" <> show (systemSeconds now) <> ".json"
+  (Qu fp) <- liftIO $ maybe (Qu <$> defFileName) return qfp
+  aq <- allQuotes
+  _ <- liftIO $ encodeFile fp aq
+  sendMessage m ("Succesfully exported all " <> (pack . show . length) aq <> " quotes to `" <> pack fp <> "`")
+
+-- | Import all the quotes in a file into the database from a given file. Superuser only.
+importQuotes :: Command
+importQuotes = Command "import" (parseComm importQ) []
+  where
+    importQ :: Quoted FilePath -> Message -> DatabaseDiscord ()
+    importQ (Qu fp) m = requirePermission Superuser m $ do
+      mqs <- liftIO $ decodeFileStrict fp
+      qs <- maybe (throwBot $ GenericException "error getting file" "there was an error obtaining or decoding the quotes json") (insertMany @Quote) mqs
+      sendMessage m ("Succesfully imported " <> (pack . show . length) qs <> " quotes")
+
+-- | Clear all the quotes from the database. Superuser only.
+clearQuotes :: Command
+clearQuotes = Command "clear" (parseComm clearQ) []
+  where
+    clearQ :: Maybe (Quoted Text) -> Message -> DatabaseDiscord ()
+    clearQ (Just (Qu "clear the quotes")) m = do
```

Comment:
This should definitely be a Superuser only command...

Reply:
Currently, even worse, an unprivileged user trying it deletes the quotes, but *doesn't save*, as it doesn't have permission to run that command...

Reply:
fixed in https://github.com/WarwickTabletop/tablebot/pull/78